dashboard/src/components/TooltipDateTime/TooltipDateTime.tsx

```diff
@@ -0,0 +1,62 @@
+import { format } from 'date-fns';
+
+import { Tooltip, TooltipContent, TooltipTrigger } from '@/components/Tooltip';
+import { getDateOffset } from '@/utils/utils';
+
+type TooltipDateTimeProps = {
+  dateTime?: Date | string;
+  forceNoDate?: boolean;
+  dateFormat?: string;
+  timeFormat?: string;
+  lineBreak?: boolean;
+  showTime?: boolean;
+  showTZ?: boolean;
+};
+
+const TooltipDateTime = ({
+  dateTime,
+  forceNoDate,
+  dateFormat,
+  timeFormat,
+  lineBreak,
+  showTime,
+  showTZ,
+}: TooltipDateTimeProps): JSX.Element => {
+  let dateObj: Date;
+
+  if (!dateTime) {
+    dateObj = new Date();
```

Comment:
this is a violation of the rules of react, it will make the component impure

https://react.dev/reference/rules

https://react.dev/reference/react/PureComponent

https://react.dev/learn/keeping-components-pure


you need to initiate this date inside a set state    `useState(date ?? new Date())`

Reply:
also this is mislieading, this component should always receive a datetime, it is not this components responsability to create a datetime

imagine if it is `undefined` in the database, it will just create a misleading datetime here

Reply:
I don't understand the logic here. If you don't receive a dateTime, will you use today as the date?

Reply:
IMO we have two better approaches in a situation that we need to do something like:
`const [currentDateTime, ] = useState<Date>(date ?? new Date())`

1. `useMemo` with an empty deps array:
`const currentDateTime = useMemo(()=> new Date(), [])`

2. `useRef` with initial value
`const currentDateTime = useRef(new Date())`